packages/rest/src/rest-http-error.ts

```diff
@@ -12,7 +13,25 @@ export namespace RestHttpErrors {
     const msg = `Parameters with "in: ${location}" are not supported yet.`;
     return new HttpErrors.NotImplemented(msg);
   }
-  export function invalidRequestBody(msg: string): HttpErrors.HttpError {
-    return new HttpErrors.UnprocessableEntity(msg);
+  export const INVALID_REQUEST_BODY_MESSAGE =
+    'The request body is invalid. See error object `details` property for more info.';
+  export function invalidRequestBody(): HttpErrors.HttpError {
+    return new HttpErrors.UnprocessableEntity(INVALID_REQUEST_BODY_MESSAGE);
   }
+  /**
+   * An invalid request body error contains a `details` property as the machine-readable error.
+   * Each entry in `error.details` contains the following attributes:
+   * - path: a path to the invalid field.
+   * - code: a single word code represents the error's type.
+   * - message: a humnan readable description of the error.
+   * - info: some additional details that the 3 attributes above don't cover.
+   * `ErrorDetails` defines the type of each entry, which is an object.
+   * The type of `error.details` is `ErrorDetails[]`
+   */
+  export type ErrorDetails = {
```

Comment:
Since these details are specific to validation errors only (and not to `invalidData` error for an example), should use a more specific type name, e.g. `ValidationErrorDetails`? Also use `export interface` instead of `export type`!